prov/sockets/src/sock_wait.c

```diff
@@ -137,8 +137,11 @@ static int sock_wait_wait(struct fid_wait *wait_fid, int timeout)
 			cq = container_of(list_item->fid,
 					  struct sock_cq, cq_fid);
 			sock_cq_progress(cq);
-			if (ofi_rbused(&cq->cqerr_rb))
+			pthread_mutex_lock(&cq->lock);
+			if (ofi_rbused(&cq->cqerr_rb)) {
+				pthread_mutex_unlock(&cq->lock);
 				return 1;
+			}
```

Comment:
There is a `pthread_mutex_lock(&cq->lock);` missing here, before the `break`.

Reply:
fixed in latest push, thanks!